mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala

```diff
@@ -299,7 +299,7 @@ class KMeans private (
       val bcCenters = sc.broadcast(centers)
 
       // Find the new centers
-      val newCenters = data.mapPartitions { points =>
+      val collected = data.mapPartitions { points =>
```

Comment:
nit: can we find a better name than `collected`?

Reply:
I am neutral on this.